assets/js/base/context/providers/cart-checkout/payment-methods/use-payment-method-registration.ts

```diff
@@ -51,18 +51,16 @@ const usePaymentMethodRegistration = (
 	const { billingData, shippingAddress } = useCustomerDataContext();
 	const selectedShippingMethods = useShallowEqual( selectedRates );
 	const paymentMethodsOrder = useShallowEqual( paymentMethodsSortOrder );
-	const {
-		cartTotals,
-		cartNeedsShipping,
-		paymentRequirements,
-	} = useStoreCart();
+	const cart = useStoreCart();
+	const { cartTotals, cartNeedsShipping, paymentRequirements } = cart;
 	const canPayArgument = useRef( {
 		cartTotals,
 		cartNeedsShipping,
 		billingData,
 		shippingAddress,
 		selectedShippingMethods,
 		paymentRequirements,
+		cart,
```

Comment:
I think this change should be a part of https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/4670, so I'd exclude it from this try branch.